app/code/core/Mage/Adminhtml/Block/Permissions/OrphanedResource/Grid.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * OpenMage
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available at https://opensource.org/license/osl-3-0-php
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2022-2023 The OpenMage Contributors (https://www.openmage.org)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * Adminhtml permissions orphanedResource grid
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ */
+class Mage_Adminhtml_Block_Permissions_OrphanedResource_Grid extends Mage_Adminhtml_Block_Widget_Grid
+{
+    public function __construct()
+    {
+        parent::__construct();
+        $this->setId('permissionsOrphanedResourceGrid');
+        $this->setDefaultSort('resource_id');
+        $this->setDefaultDir('asc');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function _prepareCollection()
+    {
+        /** @var Mage_Admin_Model_Resource_Rules_Collection */
+        $collection = Mage::getResourceModel('admin/rules_collection')
+            ->addFieldToFilter('resource_id', ['nin' => Mage::getModel('admin/roles')->getResourcesList2D()])
+            ->addFieldToSelect('resource_id');
+        $collection->getSelect()->group('resource_id');
+
+        /**
+         * In order for mass action selection to work properly, we need to overwrite
+         * the model resource $_idFieldName, from the default 'rule_id' to 'resource_id'.
+         * @see Mage_Adminhtml_Block_Widget_Grid_Massaction_Abstract::getGridIdsJson()
+         * @var Mage_Admin_Model_Resource_Rules $resource
+         */
+        $resource = $collection->getResource();
+        $resource->setResourceIdAsIdFieldName();
+
+        $this->setCollection($collection);
+        return parent::_prepareCollection();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function _prepareColumns()
```

Comment:
```suggestion
    protected function _prepareColumns(): static
```

Proposal ...

Reply:
I have not seen this before, did a quick search, is this [PHP 8](https://php.watch/versions/8.0/static-return-type)?

Reply:
Shoul be valid for 7.4.

Reply:
See [test](https://onlinephp.io/c/6d11a).. PHP7.4 throws
> Parse error: syntax error, unexpected 'static' (T_STATIC) in /home/user/scripts/code.php on line 3

Reply:
Mkay, `self` works for php74, but better leave it as it is.